packages/api/cms-api/src/generator/crud-generator.decorator.ts

```diff
@@ -1,6 +1,9 @@
 export interface CrudGeneratorOptions {
     targetDirectory: string;
-    requiredPermission?: string[] | string;
+    requiredPermission?:
+        | string[]
+        | string
+        | { list: string | string[]; create: string | string[]; update: string | string[]; delete: string | string[] };
     create?: boolean;
     update?: boolean;
     delete?: boolean;
```

Comment:
I'm not so happy with this unrelated fields. as this configuration doesn't make sense: 
```
{
    requiredPermission: { create: "createEntity, ... },
    create: false,
    update: false,
    delete: true,
}
```

Reply:
doesn't it? what do you suggest?